doc/architecture/adr-004.md

```diff
@@ -235,6 +236,12 @@ message MsgBurnNFTResponse {}
 
 To implement `MsgBurnNFT`, we can directly call the `BurnNFT` of `x/nft` (after performing validations).
 
+### Backwards incompatibility
+
+It is possible to support all the transactions and queries in a backwards compatible way except for `DenomByName` query.
+Because current implementation of NFT module specifies `denom_name` as a unique parameter unlike Cosmos SDK's native
+`x/nft` module, it is not possible to support a query which returns single `Denom` object with given `denom_name`.
```

Comment:
@kaantig is this used?

Reply:
nope, we only query it with its denom_id in the request I displayed below. we don't use denom name. 

${nodeUrl}/chainmain/nft/collections/${denomId}